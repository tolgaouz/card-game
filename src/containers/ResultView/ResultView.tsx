import React from 'react';
import WinnerSVG from '../../assets/winner.svg';
import Button from '../../components/Button';
import { CardGameState } from '../../types';

interface ResultViewProps {
  userWon?: CardGameState['userWon'];
  onPlayAgain: () => void;
}

const ResultView: React.FC<ResultViewProps> = ({ userWon = false, onPlayAgain }) => {
  return (
    <div className="my-4 md:my-0 font-courier w-full text-white flex flex-col items-center">
      {!userWon ? (
        <div data-testid="lost-text" className="w-full flex flex-col items-center">
          <div className="text-5xl md:text-6xl leading-none">Game Over</div>
          <div className="text-4xl text-center mt-4">
            You lose. <br />
            Better luck next time!
          </div>
        </div>
      ) : (
        <img
          data-testid="win-image"
          className="z-10 absolute top-36 motion-safe:animate-bounce"
          src={WinnerSVG}
          alt="Winner"
        />
      )}
      <Button
        data-testid="play-again-button"
        className="text-2xl px-9 py-2 my-3"
        onClick={() => onPlayAgain()}
        variant="outline"
      >
        Play Again
      </Button>
    </div>
  );
};

export default ResultView;
