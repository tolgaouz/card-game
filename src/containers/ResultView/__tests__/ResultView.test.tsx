import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import ResultView from '../ResultView';

describe('Result View', () => {
  it('renders game over text when userWon is false', () => {
    const { container, getByTestId } = render(
      <ResultView userWon={false} onPlayAgain={() => null} />
    );
    expect(container).toMatchSnapshot();
    expect(getByTestId('lost-text')).toBeInTheDocument();
  });

  it('successfully calls onPlayAgain', () => {
    const onPlayAgain = jest.fn();
    const { getByTestId } = render(<ResultView userWon={false} onPlayAgain={onPlayAgain} />);
    userEvent.click(getByTestId('play-again-button'));
    expect(onPlayAgain).toBeCalledTimes(1);
  });

  it('renders winner image when userWon is true', () => {
    const { container, getByTestId, queryByTestId } = render(
      <ResultView userWon onPlayAgain={() => null} />
    );
    expect(container).toMatchSnapshot();
    expect(queryByTestId('lost-text')).not.toBeInTheDocument();
    expect(getByTestId('win-image')).toBeInTheDocument();
  });
});
