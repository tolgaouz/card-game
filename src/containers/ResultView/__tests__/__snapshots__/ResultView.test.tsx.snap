// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result View renders game over text when userWon is false 1`] = `
<div>
  <div
    class="my-4 md:my-0 font-courier w-full text-white flex flex-col items-center"
  >
    <div
      class="w-full flex flex-col items-center"
      data-testid="lost-text"
    >
      <div
        class="text-5xl md:text-6xl leading-none"
      >
        Game Over
      </div>
      <div
        class="text-4xl text-center mt-4"
      >
        You lose. 
        <br />
        Better luck next time!
      </div>
    </div>
    <button
      class="border-2 text-yellow-300 border-yellow-300 font-alfa rounded-xl text-2xl px-9 py-2 my-3"
      data-testid="play-again-button"
      type="button"
    >
      Play Again
    </button>
  </div>
</div>
`;

exports[`Result View renders winner image when userWon is true 1`] = `
<div>
  <div
    class="my-4 md:my-0 font-courier w-full text-white flex flex-col items-center"
  >
    <img
      alt="Winner"
      class="z-10 absolute top-36 motion-safe:animate-bounce"
      data-testid="win-image"
      src="winner.svg"
    />
    <button
      class="border-2 text-yellow-300 border-yellow-300 font-alfa rounded-xl text-2xl px-9 py-2 my-3"
      data-testid="play-again-button"
      type="button"
    >
      Play Again
    </button>
  </div>
</div>
`;
